include/dali/core/math_util.h

```diff
@@ -53,6 +56,71 @@ DALI_HOST_DEV constexpr T clamp(const T &value, const T &lo, const T &hi) {
   return value < lo ? lo : value > hi ? hi : value;
 }
 
+DALI_HOST_DEV inline float rsqrt(float x) {
+#ifdef __CUDA_ARCH__
+  return __frsqrt_rn(x);
+#elif defined __SSE__
+  __m128 X = _mm_set_ss(x);
+  __m128 tmp = _mm_rsqrt_ss(X);
+  float y = _mm_cvtss_f32(tmp);
+  return y * (1.5f - x*0.5f * y*y);
+#else
+  int32_t i;
```

Comment:
Any reference to source of that magic?

Reply:
And justification why 2 iterations are sufficient?

Reply:
https://github.com/id-Software/Quake-III-Arena/blob/master/code/game/q_math.c#L561

Reply:
https://en.wikipedia.org/wiki/Fast_inverse_square_root
2 iterations give 6 significant digits.

Reply:
Comments added.